### ISSUE TITLE
[Feature Request] Rejection Sampling Finetuning

### ISSUE THREAD

**Issue by maksimstw** [volcengine/verl#2829]
Hi! I'd love to see support for the RAFT (Reward rAnked FineTuning) algorithm in Verl: https://arxiv.org/abs/2304.06767. It offers a simple and stable alternative to PPO by ranking multiple sampled responses with a reward model and fine-tuning on the top ones—basically alignment via supervised learning on high-reward samples.

It’d be great if Verl could include a built-in RAFT trainer (both offline and streaming would be awesome). This would make it easier to experiment with preference alignment without the complexity of full RL. Thanks!

**Comment by vermouth1992** [comment]
Actually, this can be achieved by just a few lines of changes of the existing main_ppo through hacking.
Step 1: Set adv to GRPO
Step 2: After reward computation. Set rewards of undesired samples to say -100 and desired rewards to 100
Step 3: Modify the loss to become next token prediction, which is -torch.mean(log_probs).

A more comprehensive approach is to use the rollout and trainer API and write a new main_raft.py. We will do this after we finish the model engine refactor

**Comment by Kirrito-k423** [comment]
定时清理检查结论：按社区 issue 定时处理规则关闭此 issue。

证据：最近一次非当前运行账号的公开回复或引用是 2025-07-31T04:02:13Z（actor: vermouth1992, kind: comment），早于本轮 6 个月阈值 2026-01-13T01:10:17.218155+00:00。
检查来源：https://github.com/verl-project/verl/issues/2829，comments API https://api.github.com/repos/verl-project/verl/issues/2829/comments，timeline API https://api.github.com/repos/verl-project/verl/issues/2829/timeline。

如果作者或社区成员仍能提供新的复现、证据或需求背景，请在这里重新评论请求维护者重开，或创建新的 issue 并链接到本 issue。

Scheduled cleanup conclusion: this issue is being closed under the community issue processing policy.

Evidence: The latest public reply or reference from an account other than the current automation account was 2025-07-31T04:02:13Z (actor: vermouth1992, kind: comment), which is older than the six-month threshold for this run (2026-01-13T01:10:17.218155+00:00).
Sources checked: https://github.com/verl-project/verl/issues/2829, comments API https://api.github.com/repos/verl-project/verl/issues/2829/comments, timeline API https://api.github.com/repos/verl-project/verl/issues/2829/timeline.

If the author or community can still provide a fresh reproduction, evidence, or requirement context, please comment here to ask maintainers to reopen it, or open a new issue and link back to this one.

This response was generated by a scheduled Codex GPT5.5 issue processing task.
